platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -582,45 +586,93 @@ public class WritableRosterStore extends ReadableRosterStore {
-   the desired location.
-3. DevOps will be responsible for managing the lifecycle of the files on disk that are used in creating the Roster,
-   including the new TSS encryption key.
-4. DevOps will be responsible for cleaning up the `override-config.txt` file after the network transplant process is
-   complete.
+The long pole in the tent is the lifecycle changes that impact DevOps practices. A schedule of work to minimize
+the long pole is the following implementation priority:
+
+1. Roster Data Structure and State Management
+2. Roster Lifecycle gated behind a feature flag
+3. Documentation of the new roster life-cycle API for genesis and network transplant.
+4. Duplication and modification of existing DAB tests to use Roster Lifecycle.
+5. Development of new HAPI and JRS Tests
+6. Refactor of the platform to use `Roster` instead of `AddressBook`.
+7. Code Cleanup and Implementation Documentation
+
+Manual Testing Plan:
+
+* Once `3.` is completed above, it becomes possible to add testing labor to manually test the new startup logic on our
+  test networks.
+
+### DevOps Process Changes, in summary
+
+#### Genesis Network
+
+1. There must be no state present on disk to load.
+2. Provide a `genesis-config.txt` file on disk in the same directory that `config.txt` is normally provided. The
+   file format is the same as config.txt.
+3. After startup, the `genesis-config.txt` file will be moved to a subdirectory `archive/yyyy-MM-dd_HH-mm-ss/`
+   to indicate that it has been used.
+
+#### Normal Restart
+
+1. There must be a loadable state present on disk in the standard location.
+2. There must not be a software upgrade happening.
+3. Ensure that `config.txt`, `genesis-config.txt`, and `override-config.txt` are not present on disk to avoid a
+   warnings in the log files. These files will be ignored.
+
+#### Software Upgrade (No Transplant, First Upgrade With New Roster Lifecycle)
+
+1. There must be a loadable state present on disk in the standard location.
+2. There must be a software upgrade happening.
+3. Provide the last `config.txt` file on disk in the same directory that `config.txt` is normally provided.
+   * Following the existing DAB procedure is expected here.
+4. Do not provide a `genesis-config.txt` or `override-config.txt` file on disk.
+5. After startup the `config.txt` file will be moved to a subdirectory `archive/yyyy-MM-dd_HH-mm-ss/`.
+6. After this release there will no longer be a `config.txt` generated on freeze upgrade shutdown.
+
+#### Software Upgrade (No Transplant, Subsequent Upgrades)
+
+1. There must be a loadable state present on disk in the standard location.
+2. There must be a software upgrade happening.
+3. Ensure that `config.txt`, `genesis-config.txt`, and `override-config.txt` are not present on disk.
+
+#### Software Upgrade Network Transplant
+
+1. Copy the PCES files from a single node in the source network and replicate to all nodes in the target network.
```

Comment:
Minor q: does this also assume that a state **from a single node** (probably, the same single node) will be replicated to all the nodes as well? If so, let's add a point about this for clarity.

If not, then I'm wondering how reliable is it to use PCES files from a single node with potentially different states on different nodes? I'd guess this may cause some flakiness, and I'm not sure if there's a good reason to tolerate this flakiness.

Reply:
If the states are different, an ISS would result.   There is a known event branching issue when we copy multiple PCES files from different nodes.   But I will add the text about using the same state file.

Reply:
Sounds good. Thanks for adding the text!